Sources/Layout/FlowLayout.swift

```diff
@@ -26,21 +26,51 @@ public class FlowLayout<Data>: AxisDependentLayout<Data> {
   public override func layout(collectionSize: CGSize,
                               dataProvider: CollectionDataProvider<Data>,
                               sizeProvider: @escaping CollectionSizeProvider<Data>) -> [CGRect] {
+    guard dataProvider.numberOfItems > 0 else {
+      return []
+    }
     var frames: [CGRect] = []
     var primaryOffset: CGFloat = 0
     var secondaryOffset: CGFloat = 0
     var currentMaxPrimary: CGFloat = 0
-    for i in 0..<dataProvider.numberOfItems {
-      let size = sizeProvider(i, dataProvider.data(at: i), collectionSize)
-      if secondaryOffset + secondary(size) > secondary(collectionSize), secondaryOffset != 0 {
+    var i = 0
+    while i < dataProvider.numberOfItems {
+      var currentFitSizes: [CGSize] = []
+      for j in i..<dataProvider.numberOfItems {
+        let size = sizeProvider(j, dataProvider.data(at: j), collectionSize)
```

Comment:
Ideally we shouldn't call sizeProvider twice when doing layout. if the cell is a composer. then that needs to be layout twice